src/sharding/ShardingManager.js

```diff
@@ -44,16 +44,33 @@ class ShardingManager extends EventEmitter {
     if (!stats.isFile()) throw new Error('CLIENT_INVALID_OPTION', 'File', 'a file');
 
     /**
-     * Amount of shards that this manager is going to spawn
-     * @type {number|string}
+     * List of shards this sharding manager spawns
+     * @type {string|number[]}
      */
-    this.totalShards = options.totalShards;
+    this.shardList = options.shardList || 'auto';
+    if (this.shardList !== 'auto') {
+      if (!Array.isArray(this.shardList)) {
+        throw new TypeError('CLIENT_INVALID_OPTION', 'Shardlist', 'an array.');
+      }
+      this.shardList = [...new Set(this.shardList)];
+      if (this.shardList.length < 1) throw new RangeError('CLIENT_INVALID_OPTION', 'Shardlist', 'at least 1 ID.');
+      if (this.shardList.some(shardID => typeof shardID !== 'number' || isNaN(shardID) ||
+        !Number.isInteger(shardID) || shardID < 0)) {
+        throw new TypeError('CLIENT_INVALID_OPTION', 'Shardlist', 'an array of postive integers.');
```

Comment:
`Shardlist` -> `shardList` to be more congruent with the client option capitalization.